util/expensivequery/expensivequery.go

```diff
@@ -73,12 +92,145 @@ func (eqh *Handle) Run() {
 				}
 			}
 			threshold = atomic.LoadUint64(&variable.ExpensiveQueryTimeThreshold)
+
+			instanceStats := &runtime.MemStats{}
+			runtime.ReadMemStats(instanceStats)
+			instanceMem := instanceStats.HeapAlloc
```

Comment:
We'd better use `MemStats.NextGC` to decide do we need to record memory usage information.

Go's GC don't take the system memory limit into consideration, that means:
* You have 16G physical memory;
* You have 9G live objects;
* Go will run GC when memory usage reaches `9G * (1 + GOGC / 100) = 18G`, OOM.

`runtime.ReadMemStats` and pprof's memory profile only update after GC has finished. So before memory usage reaches 18G or 2m has passed after the previous GC, you will know nothing about heap usage.

BTW, the pprof's memory profile is not a real-time snapshot of the heap, it is a snapshot of the heap when the penultimate GC has finished. You can see more detailed information at https://github.com/golang/go/blob/ca978a5ca9390dc759f20db989b49ad9cdcd81d4/src/runtime/mprof.go#L60